src/test/shared/sam/cli/samCliBuild.test.ts

```diff
@@ -59,17 +59,25 @@ describe('SamCliBuildInvocation', async () => {
         await del([tempFolder], { force: true })
     })
 
-    it('Passes build command to sam cli', async () => {
+    it('invokes `sam build` with args', async () => {
         const processInvoker: SamCliProcessInvoker = new ExtendedTestSamCliProcessInvoker((args: any[]) => {
-            assert.ok(args.length > 0, 'Expected args to be present')
-            assert.strictEqual(args[0], 'build', 'Expected first arg to be the build command')
+            assert.ok(args.length >= 2, 'Expected args to be present')
+            assert.strictEqual(args[0], 'build')
+            assert.strictEqual(args[3], '--template')
+            assert.strictEqual(args[5], '--base-dir')
+
+            // `extraArgs` are appended to the end.
+            assert.strictEqual(args[7], '--debug')
+            assert.strictEqual(args[8], '--build-dir')
+            assert.strictEqual(args[9], 'my/build/dir/')
         })
```

Comment:
todo (future): most of the tests in this file are trivial variations on the above; could avoid a lot of code by putting them in 1 test ...